adapters/horizondex/src/sdk/positionSnapshots.ts

```diff
@@ -0,0 +1,167 @@
+            orderBy: blockNumber, 
+            orderDirection: asc,
+            where: {
+              blockNumber_gt: ${b_start},
+              blockNumber_lte: ${b_end},
+            }
+          ) {
+               pool {
+                id
+                token0 {
+                id
+                symbol
+                }
+                token1 {
+                id
+                symbol
+                }
+              }
+              depositedToken0
+              depositedToken1
+              withdrawnToken0
+              withdrawnToken1
+              owner
+              liquidity
+              blockNumber
+              timestamp
+          }
+        }
+  `
+        const res = await fetch(SUBGRAPH_URLS[CHAINS.LINEA],{
+            method: "POST",
+            body: JSON.stringify({ query }),
+            headers: { "Content-Type": "application/json" },
+        }).then(res => res.json()) as SubgraphResponse
+
+        snapshotsArrays = snapshotsArrays.concat(res.data.positionSnapshots)
+        if(res.data.positionSnapshots.length == 0) {
+            break
+        }
+        skip += 1000
+        writeProgress(b_end, b_start,b_end)
+    }
+    for(const snapshot of snapshotsArrays) {
+        let userPositionSnapshotMap = snapshotsMap.get(snapshot.owner)
+        if(!userPositionSnapshotMap) {
+            userPositionSnapshotMap = new Map<string, PositionSnapshot>()
+        }
+        userPositionSnapshotMap.set(snapshot.pool.id, snapshot)
+        snapshotsMap.set(snapshot.owner, userPositionSnapshotMap)
+        if (snapshot.timestamp > maxTimestamp) {
+            maxTimestamp = snapshot.timestamp
+        }
+    }
+
+    //getting token pricing snapshot for positions usd values
+    const tokenPricingSnapshots = await getTokenPricingSnapshots(maxTimestamp)
+
+    snapshotsMap.forEach((userPositionSnapshotMap => {
+        userPositionSnapshotMap.forEach((positionSnapshot) => {
+            const token0UsdSnapshot = STABLES.includes(positionSnapshot.pool.token0.id) ? {
+                priceUSD: "1.0",
+                periodStartUnix: "0",
+                token: {
+                    id: positionSnapshot.pool.token0.id
+                }
+            } : findClosestTokenSnapshot(tokenPricingSnapshots.get(positionSnapshot.pool.token0.id), positionSnapshot.timestamp)
+            const token1UsdSnapshot = STABLES.includes(positionSnapshot.pool.token1.id) ? {
+                priceUSD: "1.0",
+                periodStartUnix: "0",
+                token: {
+                    id: positionSnapshot.pool.token1.id
+                }
+            } : findClosestTokenSnapshot(tokenPricingSnapshots.get(positionSnapshot.pool.token1.id), positionSnapshot.timestamp)
+            const token0Amount = parseFloat(positionSnapshot.depositedToken0) - parseFloat(positionSnapshot.withdrawnToken0);
+            const token1Amount = parseFloat(positionSnapshot.depositedToken1) - parseFloat(positionSnapshot.withdrawnToken1);
```

Comment:
this wont be accurate, token balances are affected by swaps as well.

i see that you guys are CLMM dex, can you refer to sushi's code as reference?

we only want to track active positions (position that are in range)

https://github.com/delta-hq/l2-lxp-liquidity-reward/pull/37/files